parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -228,34 +183,26 @@ pub mod pallet {
 				Error::<T>::PayloadTooLarge,
 			);
 
-			let next_id = <NextId<T>>::get();
-			if next_id.checked_add(1).is_none() {
-				return Err(Error::<T>::Overflow.into())
-			}
+			let nonce = <Nonce<T>>::get(source_id);
+			let next_nonce = nonce.checked_add(1).ok_or(Error::<T>::Overflow)?;
 
-			<MessageQueue<T>>::try_append(EnqueuedMessage {
-				account: who.clone(),
-				message: Message {
-					id: next_id,
-					target,
-					payload: payload
-						.to_vec()
-						.try_into()
-						.map_err(|_| Error::<T>::PayloadTooLarge)?,
-				},
+			<MessageQueue<T>>::try_append(Message {
+				source_id: source_id.clone(),
+				nonce,
+				payload: payload.to_vec().try_into().map_err(|_| Error::<T>::PayloadTooLarge)?,
```

Comment:
seems `PayloadTooLarge` is already checked in previous step, so `try_into` not nessessary here?

Reply:
Yes, you're right. I'll move this conversion into the `ensure!` above.